State Records Office of Western Australia.js

```diff
@@ -0,0 +1,372 @@
+	}
+
+	if (item.title === "") {
+		item.title = "[Title Not Found]";
+	}
+
+	item.url = url;
+	
+	var contentArea = doc.querySelector("section#contentAndStructureArea");
+	
+	if (contentArea) {
+		var contentFields = contentArea.querySelectorAll("div.field");
+	
+		for (var j=0;j<contentFields.length;j++) {	
+			
+			nextContentField = text(contentFields[j],"h3");
+			
+			switch(nextContentField)
+			{
+				case "Scope and content":
+					item.abstractNote = ZU.trimInternal(text(contentFields[j],"div"));
+					break;
+
+				default:
+					if (item.abstractNote === "") {
+						item.abstractNote = "";
+					}
+					break;
+			}
+	
+		}			
+	}
+
+	var contextArea = doc.querySelector("section#contextArea");
+	
+	// all SROWA creators are organisations not individuals ; therefore remove the first part of the
+	// creator reference code eg. "AU WA A58 - ";
+
+	if (contextArea) {
+		var contextFields = contextArea.querySelectorAll("div.field");
+		
+		for (var k=0;k<contextFields.length;k++) {	
+			
+			nextContextField = text(contextFields[k],"h3");
+			
+			switch(nextContextField)
+			{
+				case "Name of creator":
+					creatorName = text(contextFields[k],"div > a").split(" - ");
+					item.creators.push({
+						lastName: creatorName[1],
+						creatorType: "author",
+						fieldMode: 1
+					})
+					break;
+
+				default:
+					if (item.creatorName === '') {
+						item.creatorName = '';
+					}
+			}
+	
+		}	
+		
+	}
+	
+	// attachment(s)
+	
+	var digitalObject = doc.querySelector("div.digital-object-reference");
+	
+	if (digitalObject) {
+		item.attachments.push({
+			title : '',
+			url : attr(digitalObject,'a','href'),
+			mimeType : selectMimeType(attr(digitalObject,'a','href')),
+			snapshot : true
+		});
+	}
+		
+	item.complete();
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://archive.sro.wa.gov.au/index.php/northampton-classification-4-tally-no-000782-northampton-04-1",
+		"items": [
+			{
+				"itemType": "manuscript",
+				"title": "Northampton, Classification 4 [Tally No. 000782].",
+				"creators": [
+					{
+						"lastName": "DEPARTMENT OF LANDS AND SURVEYS",
+						"creatorType": "author",
+						"fieldMode": 1
+					}
+				],
+				"date": "1906-01-17",
+				"archiveLocation": "AU WA S980- cons4936 Northampton 04.1",
+				"libraryCatalog": "State Records Office of Western Australia",
+				"manuscriptType": "Item",
+				"url": "https://archive.sro.wa.gov.au/index.php/northampton-classification-4-tally-no-000782-northampton-04-1",
+				"attachments": [],
+				"tags": [],
+				"notes": [
+					{
+						"title": "Date(s)",
+						"note": "Date(s): - 1906-01-17 (Accumulation)"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://archive.sro.wa.gov.au/index.php/cancelled-public-plans-districts-post-standard-series-s980",
+		"items": [
+			{
+				"itemType": "manuscript",
+				"title": "CANCELLED PUBLIC PLANS (DISTRICTS) - POST STANDARD SERIES",
```

Comment:
question: we can try to convert these all caps strings to something more sensible -- would that make sense or is it important that they stay in all caps for some reason?

Reply:
Thank you for improving my javascript. I will:
1. Review that it still works as expected.
2. Add some code to convert the caps to Title case or similar (SROWA have OK'ed that).
3. Add the tests.

Also, is ZU.Xpath still acceptable in translators?  I went with DOM methods eg. querySelectorAll(), but this may have added

Reply:
We prefer using CSS selectors over xpaths for new translators if at all possible, yes, so that was the right call.